app/services/vot_parser.rb

```diff
@@ -0,0 +1,68 @@
+class VotParser
+  class VotParseException < StandardError; end
+
+  VotParserResult = Data.define(:identity_proofing, :credential_usage)
+
+  VECTOR_COMPONENT_REGEXP = /\A(?<component>[A-Z])(?<value>[0-9a-z])\z/
+
+  IDENTITY_PROOFING_VECTOR_VALUES = {
+    '0' => :no_identity_proofing,
+    '1' => :identity_proofing_no_biometric,
+    '2' => :identity_proofing_biometric_required,
+  }.freeze
+  CREDENTIAL_USAGE_VECTOR_VALUES = {
+    '0' => :default,
+    '1' => :no_remember_device,
+    '2' => :unphishable_mfa,
+    '3' => :piv_cac_required,
+  }.freeze
+
+  attr_reader :vector_of_trust
+
+  def initialize(vector_of_trust)
+    @vector_of_trust = vector_of_trust
+  end
+
+  def parse
+    result = { identity_proofing: :no_identity_proofing, credential_usage: :default }
+    vector_component_map.each do |component, value|
+      if component == 'P'
+        result[:identity_proofing] = read_vector_value_from_map(
+          component: 'P',
+          value: value,
+          map: IDENTITY_PROOFING_VECTOR_VALUES,
+        )
+      elsif component == 'C'
+        result[:credential_usage] = read_vector_value_from_map(
+          component: 'C',
+          value: value,
+          map: CREDENTIAL_USAGE_VECTOR_VALUES,
+        )
+      else
+        raise VotParseException, "#{vector_of_trust} contains unsupported component #{component}"
+      end
```

Comment:
might be a good opportunity for case/when, could even use the new left-side-only hash syntax to remove some repeated parts

```suggestion
      case component
      when 'P'
        result[:identity_proofing] = read_vector_value_from_map(
          component:,
          value:,
          map: IDENTITY_PROOFING_VECTOR_VALUES,
        )
      when 'C'
        result[:credential_usage] = read_vector_value_from_map(
          component:,
          value:,
          map: CREDENTIAL_USAGE_VECTOR_VALUES,
        )
      else
        raise VotParseException, "#{vector_of_trust} contains unsupported component #{component}"
      end
```